### DISCUSSION TITLE
Lesson 7: Transfer function NOT found in LinkToken.sol function call (Video Ref: 7:41:29)

### DISCUSSION THREAD

**Discussion by abdullahpichoo** [smartcontractkit/full-blockchain-solidity-course-py#1666] [Q&A] [answered]
## Error: 
<img width="1006" alt="image" src="https://user-images.githubusercontent.com/98353306/186538441-0a1406f3-59e5-4f25-ac90-b61008b7e1f4.png">

## Code ``fund_with_link()``
```Python
def fund_with_link(contract_address, acc=None, link_token=None, amount=LINK_AMOUNT):
    if acc:
        acc = acc
    else:
        acc = get_acc

    if link_token:
        link_token = link_token
    else:
        link_token = get_contract("link-token")
    # 7.43.48 Interface working example to replace Line 55
    # link_token_contract = interface.LinkTokenInterface(link_token.address)
    print(link_token.totalSupply())
    print(link_token.name())
    tx = link_token.transfer(contract_address, amount, {"from": acc})
    tx.wait(1)
    print("Contract Funded with Link")
    return tx
```

## Problem
Hi! I was working with the Lottery Project and was coding the ```fund_with_link``` function. Line 86 seems to be causing this problem as it says the LinkToken.sol file has no function named ``transfer`` but I've checked the sol file and ofc it's there. I've also tried doing it with the interface example but that issue still seems to be there.

I thought the line 81 ```link_token = get_contract("link-token")``` might not be returning the right contract but the calls I made in line 84 and line 85 ```print(link_token.totalSupply())``` ```print(link_token.name())``` seem to return the correct values from the LinkToken.sol contract.

The issue I'm getting is with this transfer function and my guess is that the problem might be in that line 42 of LinkToken.sol contract that I'm attaching a snippet of, here:
<img width="486" alt="image" src="https://user-images.githubusercontent.com/98353306/186539675-f656a213-5205-4176-a2bb-03f0c98bde45.png">

I really have no idea of what to do here, I would appreciate some help on this.

**Comment by cromewar** [comment, marked as answer]
Hello @abdullahpichoo 
This is weird as you are 100% right, Link token do have transfer function.
However you have another error, on your get_acc you are not including the parenthesis, it should look like this `get_acc()`, also try deleting the build folder and try again.
Please post any update on this I'll try to help.

**Reply by abdullahpichoo** [reply]
Thanks a lot man! Those parentheses had me tripping for like an hour until I gave up and posted here. One more question tho, is there a way to debug these functions like we do in Visual Studio with breakpoints and step into, etc. I'm not sure if we can debug step by step with brownie. I've coded a lot in Visual Studio and debugging has saved me a ton of time but is there a way we can follow a similar approach with brownie?

Thanks again!!

**Reply by cromewar** [reply]
Yes you can, when you run a test with `brownie test` you can add the flag `-s` to get a step to step execution.
so your you should run something like this: `brownie test -s -k name_of_test`.
I hope this info might help.
Cheers.
